src/main/java/com/ust/sourcecourse/configuration/service/DBDataSourceService.java

```diff
@@ -135,22 +135,37 @@ public ResponseEntity<String> removeTagFromSourceTable(Long uid, String tag) {
 
 	public List<DBTable> searchTablesByTag(String tag) {
 		List<SourceTable> sourcetables = sourceTableRepository.findAll();
-		return sourcetables.stream()
-				.filter(sourcetable -> sourcetable.getTags() != null && sourcetable.getTags().contains(tag))
-				.map(sourcetable -> getDBTable(sourcetable)).collect(Collectors.toList());
+		if (sourcetables.isEmpty()) {
+			throw new CustomException("No tables found in the database.");
```

Comment:
If it's empty, the return status should be 201-NO_CONTENT. Either handle in controller and return ResponseEntity.noContent() or have another custom exception for NoContent and throw it here and in Exception Handler, return no content status for that exception.

Reply:
resolved